compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala

```diff
@@ -34,7 +34,7 @@ class QuoteDriver(appClassloader: ClassLoader) extends Driver {
     val ctx = setToolboxSettings(ctx0.fresh.setSetting(ctx0.settings.outputDir, outDir), settings)
 
     val driver = new QuoteCompiler
-    driver.newRun(ctx).compileExpr(expr)
+    driver.newRun(ctx).compileExpr(exprBuilder)
```

Comment:
It seems after `driver.newRun(ctx)`, we already be able to create an instance of `QuoteContext` to call `exprBuilder()`, such that we can keep `ExprCompilationUnit` and related API unchanged as before?

Reply:
We do have a context but it is not the correct one. In `compileExpr` we call [`compileUnits`](https://github.com/lampepfl/dotty/blob/946500136434d41aa6e5b55dcfaa3c1bd2589fbd/compiler/src/dotty/tools/dotc/Run.scala#L133) which setup the correct context . I would not dare duplicating this logic here.